app/assets/javascripts/application.js

```diff
@@ -20,4 +20,10 @@
 //= require nprogress
 //= require nprogress-turbolinks
 //= require main
-//= require jquery.sidr.min
\ No newline at end of file
+//= require jquery.sidr.min
+
+$(document).ajaxError(function (e, xhr, settings) {
```

Comment:
> Note: This handler is not called for cross-domain script and cross-domain JSONP requests.

Oh, ok - that solves most of our problems. When I said "created by Rails", I meant created by [ujs](https://github.com/rails/jquery-ujs).

Looks like jquery-ujs does publish events for errors that we can use: 
https://github.com/rails/jquery-ujs/wiki/ajax.

It's just that - I'd like this to look less like magic. When we use `data-remote: true`, we're aware that we're using a function in Rails so it's okay to expect 401s to be handled for us. If someone ends up writing an AJAX call by hand, they shouldn't be surprised by the automatic 401 patch, they should be explicitly mentioning it in the error handler for that request. Two requests, one created by rails and one written by hand could end up calling the same error function, but we shouldn't assume that every request would need the 401 patch. Makes sense?